### PR TITLE
[ISSUE#4515] Add test case for ProtocolPluginFactory.

### LINKED ISSUE #4515: [Enhancement] Add test case for ProtocolPluginFactory.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

Add test case for ProtocolPluginFactory.

### Describe the solution you'd like

Add test case for ProtocolPluginFactory.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct) *

### PR DESCRIPTION
**Pull request by yanrongzhen**
Fixes #4515 

### Modifications
Add test case for ProtocolPluginFactory.

### CODE REVIEW

File: eventmesh-protocol-plugin/eventmesh-protocol-api/src/test/resources/META-INF/eventmesh/org.apache.eventmesh.protocol.api.ProtocolAdaptor [hunk 1]
```diff
@@ -0,0 +1 @@
+testProtocolType=org.apache.eventmesh.protocol.api.MockProtocolAdaptorImpl
```
Comment (Alonexc) [inline eventmesh-protocol-plugin/eventmesh-protocol-api/src/test/resources/META-INF/eventmesh/org.apache.eventmesh.protocol.api.ProtocolAdaptor:1]: A license needs to be added here.

Comment (yanrongzhen) [reply eventmesh-protocol-plugin/eventmesh-protocol-api/src/test/resources/META-INF/eventmesh/org.apache.eventmesh.protocol.api.ProtocolAdaptor:17]: Done.

File: eventmesh-protocol-plugin/eventmesh-protocol-api/src/test/java/org/apache/eventmesh/protocol/api/ProtocolPluginFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        ProtocolAdaptor<ProtocolTransportObject> actualAdaptor = ProtocolPluginFactory.getProtocolAdaptor(PROTOCOL_TYPE_NAME);
+        Assertions.assertEquals(expectedAdaptor, actualAdaptor);
+    }
+
+    @Test
+    public void testGetProtocolAdaptorWhenMapNotEmpty() {
+        ProtocolAdaptor<ProtocolTransportObject> adaptor = ProtocolPluginFactory.getProtocolAdaptor(PROTOCOL_TYPE_NAME);
+        Assertions.assertEquals(adaptor.getClass(), MockProtocolAdaptorImpl.class);
+    }
```
[the remark this answers is not in the document]
Comment (yanrongzhen) [reply eventmesh-protocol-plugin/eventmesh-protocol-api/src/test/java/org/apache/eventmesh/protocol/api/ProtocolPluginFactoryTest.java:61]: It has been changed to a clearer name.

File: eventmesh-protocol-plugin/eventmesh-protocol-api/src/test/java/org/apache/eventmesh/protocol/api/ProtocolPluginFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            new ConcurrentHashMap<>(16);
+        ProtocolAdaptor<ProtocolTransportObject> expectedAdaptor = new MockProtocolAdaptorImpl();
+        mockProtocolAdaptorMap.put(PROTOCOL_TYPE_NAME, expectedAdaptor);
+
+        Field field = ProtocolPluginFactory.class.getDeclaredField("PROTOCOL_ADAPTOR_MAP");
+        field.setAccessible(true);
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
+        field.set(null, mockProtocolAdaptorMap);
```
[the remark this answers is not in the document]
Comment (yanrongzhen) [reply eventmesh-protocol-plugin/eventmesh-protocol-api/src/test/java/org/apache/eventmesh/protocol/api/ProtocolPluginFactoryTest.java:51]: Although it does not affect normal use, I handled this with @AfterEach.

File: eventmesh-protocol-plugin/eventmesh-protocol-api/src/test/java/org/apache/eventmesh/protocol/api/ProtocolPluginFactoryTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (yanrongzhen) [reply eventmesh-protocol-plugin/eventmesh-protocol-api/src/test/java/org/apache/eventmesh/protocol/api/ProtocolPluginFactoryTest.java:63]: Done

**Pull request merged**